core/sql/executor/cluster.cpp

```diff
@@ -2395,7 +2395,7 @@ NABoolean Cluster::checkAndSplit(ExeErrorCode * rc)
 				         rc);
   
   if ( !next_ || *rc ) {
-    if ( ! rc ) * rc = EXE_NO_MEM_TO_EXEC;
+    * rc = EXE_NO_MEM_TO_EXEC;
```

Comment:
The deleted code is correct.  In the new code rc could be NULL

Reply:
if rc is NULL,  core at line 2397
so rc must be not NULL.

or
make sure rc is not NULL at line 2397

Reply:
Then the condition should be
  if (!next_ || rc) 
     if ( !rc ) *rc = EXE_NO_MEM_TO_EXEC;